virtool/config.py

```diff
@@ -159,6 +203,31 @@ def start_server(ctx, host, port, no_check_db, no_check_files, no_client, no_fet
         port=port
     )
 
+    if use_b2c:
+        if config.port != 9950:
+            logger.fatal("registered endpoints for Azure AD B2C require server to run on port 9950")
+            sys.exit(1)
+        if not b2c_client_id or not b2c_client_secret or not b2c_tenant or not b2c_user_flow:
+            logger.fatal("Required B2C client information not provided. \n--use-b2c option requires the following:"
+                         "\n--b2c-client-id"
+                         "\n--b2c-client-secret"
+                         "\n--b2c-tenant"
+                         "\n--b2c-user-flow\n")
+            sys.exit(1)
+
+        msal_app = msal.ClientApplication(
+            client_id=b2c_client_id,
+            authority=f"https://{b2c_tenant}.b2clogin.com/{b2c_tenant}.onmicrosoft.com/{b2c_user_flow}",
+            client_credential=b2c_client_secret,
+        )
+
+        os.environ["B2C_CLIENT_ID"] = b2c_client_id
+        os.environ["B2C_CLIENT_SECRET"] = b2c_client_secret
+        os.environ["B2C_AUTHORITY"] = f"https://{b2c_tenant}.b2clogin.com/{b2c_tenant}.onmicrosoft.com/{b2c_user_flow}"
```

Comment:
Why? This seems a bad practice to me. An application shouldn't be overriding environmental variables potentially set in the shell.

Reply:
OK, I just thought it would be an easy way to update these values on startup every time, I'll take this out and just check that those values are still there without passing them in.